### ISSUE TITLE
Bug: file in eslint.config.js not respected

### ISSUE THREAD

**Issue by cr3** [eslint/eslint#19470]
### Environment

Node version: 22.14.0
npm version: 10.9.2
Local ESLint version: 9.21.0
Global ESLint version: 9.21.0
Operating System: Debian

### What parser are you using?

Default (Espree)

### What did you do?

I ran the `eslint .` command with the configuration below.
<details>
<summary>Configuration</summary>

```js
import js from '@eslint/js'
import globals from 'globals'
import reactHooks from 'eslint-plugin-react-hooks'
import reactRefresh from 'eslint-plugin-react-refresh'
import tseslint from 'typescript-eslint'

export default tseslint.config(
  {
    extends: [js.configs.recommended, ...tseslint.configs.recommended],
    files: ['src/**/*.{ts,tsx}'],
    languageOptions: {
      ecmaVersion: 2020,
      globals: globals.browser,
    },
    plugins: {
      'react-hooks': reactHooks,
      'react-refresh': reactRefresh,
    },
    rules: {
      ...reactHooks.configs.recommended.rules,
      'react-refresh/only-export-components': [
        'warn',
        { allowConstantExport: true },
      ],
    },
  },
)
```
</details>

```js
<!-- Paste your code here -->
```

### What did you expect to happen?

The expected behavior is to only lint .ts and .tsx files under the src directory.

### What actually happened?

It actually lints files in all other directories too. 

### Link to Minimal Reproducible Example

http://google.ca

### Participation

- [ ] I am willing to submit a pull request for this issue.
